codegen/src/lut.rs

```diff
@@ -0,0 +1,470 @@
+                    (linear_scale * linear_end - 1.0) / (linear_end.powf(gamma.recip()) - 1.0),
+                    linear_end,
+                )
+            } else {
+                (None, 1.0, 0.0)
+            };
+        Self {
+            fn_type: format_ident!("{fn_type}"),
+            fn_type_uppercase: fn_type_uppercase.to_owned(),
+            into_linear: Box::new(move |encoded| match linear_scale {
+                Some(scale) if encoded <= scale * beta => encoded / scale,
+                _ => ((encoded + alpha - 1.0) / alpha).powf(gamma),
+            }),
+            linear_scale,
+            alpha,
+            beta,
+            gamma,
+        }
+    }
+}
+
+fn build_u8_to_f64_lut(entries: &[LutEntryU8]) -> TokenStream {
+    let tables = entries.iter().map(
+        |LutEntryU8 {
+             fn_type,
+             fn_type_uppercase,
+             into_linear,
+             ..
+         }| {
+            let table = (0..=u8::MAX).map(|i| into_linear((i as f64) / 255.0));
+            let table_ident = format_ident!("{fn_type_uppercase}_U8_TO_F64");
+            quote! {
+                const #table_ident: [f64; 256] = [
+                    #(#table),*
+                ];
+
+                impl IntoLinear<f64, u8> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u8) -> f64 {
+                        #table_ident[encoded as usize]
+                    }
+                }
+
+                impl IntoLinear<f32, u8> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u8) -> f32 {
+                        #table_ident[encoded as usize] as f32
+                    }
+                }
+            }
+        },
+    );
+
+    quote! {
+        #(#tables)*
+    }
+}
+
+fn build_u16_to_f64_lut(entries: &[LutEntryU16]) -> TokenStream {
+    let tables = entries.iter().map(
+        |LutEntryU16 {
+             fn_type,
+             fn_type_uppercase,
+             into_linear,
+             ..
+         }| {
+            let table = (0..=u16::MAX).map(|i| into_linear((i as f64) / 65535.0));
+            let table_ident = format_ident!("{fn_type_uppercase}_U16_TO_F64");
+            quote! {
+                #[cfg(feature = "gamma_lut_u16")]
+                static #table_ident: [f64; 65536] = [
+                    #(#table),*
+                ];
+
+                #[cfg(feature = "gamma_lut_u16")]
+                impl IntoLinear<f64, u16> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u16) -> f64 {
+                        #table_ident[encoded as usize]
+                    }
+                }
+
+                #[cfg(feature = "gamma_lut_u16")]
+                impl IntoLinear<f32, u16> for #fn_type {
+                    #[inline]
+                    fn into_linear(encoded: u16) -> f32 {
+                        #table_ident[encoded as usize] as f32
+                    }
```

Comment:
The same thing here as for `u8`. I wonder if it's also worth having a `u16` to `f32` table here. Let's see if it helps the `u8` case.